src/main/kotlin/com/nextstep/jngcii/minesweeper/domain/MineMap.kt

```diff
@@ -0,0 +1,28 @@
+package com.nextstep.jngcii.minesweeper.domain
+
+@JvmInline
+value class MineMap private constructor(val rows: List<Row>) {
+    companion object {
+        fun build(locations: Locations): MineMap {
+            val rows = getInitialTwoDimensionalList(
+                locations.meta.rowCount,
+                locations.meta.columnCount
+            )
+
+            locations.pairs.forEach {
+                rows[it.y][it.x] = locations.check(it.x, it.y)
+            }
+
+            return MineMap(rows.map { Row(it) })
```

Comment:
객체 생성은 생성자에게 맡겨보면 어떨까요?
그런데 타입삭제때문에 `Row` 클래스에 부 생성자를 만들기 어려운 상황이네요. 기본생성자를 사용하는게 가장 좋고 기본생성자를 사용할 수 없는경우에는 팩토리 함수를 사용하는게 좋을것 같습니다.
```suggestion
            return MineMap(Row.from(rows))
```

Reply:
@karian7 저도... 저거(`List<List<Boolean>>` 조금 불편했어요.
[로직 리팩터링](https://github.com/next-step/kotlin-minesweeper/commit/39fc4a65217b138db01c73b7a5eaea0a5f4502e1)되면서 저부분이 사라지긴 했지만,
이중제네릭 하나를 위해 일급컬렉션을 만드는 것도 조금 과한것같고 `typealias`를 사용하면 가독성이 조금 좋아질까요? (아니면 더 헷갈리게 만드는 걸까요)

Reply:
typealias 도 좋지만 저는 이런 접근을 제안해요.
컬렉션을 만든이상 이 컬렉션을 가지고 무언가를 하는 부분이 분명 있을꺼에요. 리스트에서 뭔가 가져오거나 리스트를 기반으로 조작하거나.. 따라서 일급컬렉션을 만들고 이 컬렉션을 사용하는곳을 유심히 살펴셔 추가적인 책임을 일급컬렉션에게 위임하는 시도를 해보는거에요.
그런데 정말로 이 리스트를 만들어서 전~~혀 손대지 않는다면(예를들면 바로 직렬화 해버린다던가..) 그냥 둬도 되지 않을까 해요.ㅎㅎ